src/Surfnet/StepupMiddleware/CommandHandlingBundle/SensitiveData/Repository/SensitiveDataMessageRepository.php

```diff
@@ -64,25 +65,53 @@ public function findByIdentityId(IdentityId $identityId)
     }
 
     /**
-     * @param SensitiveDataMessage[] $sensitiveDataMessages
+     * @param SensitiveDataMessageStream $sensitiveDataMessageStream
      * @return void
      */
-    public function append(array $sensitiveDataMessages)
+    public function append(SensitiveDataMessageStream $sensitiveDataMessageStream)
     {
         $this->connection->beginTransaction();
 
         try {
-            foreach ($sensitiveDataMessages as $sensitiveDataMessage) {
+            foreach ($sensitiveDataMessageStream as $sensitiveDataMessage) {
+                /** @var SensitiveDataMessage $sensitiveDataMessage */
                 $this->connection->insert('event_stream_sensitive_data', [
                     'identity_id'    => (string) $sensitiveDataMessage->getIdentityId(),
                     'playhead'       => $sensitiveDataMessage->getPlayhead(),
-                    'sensitive_data' => json_encode($sensitiveDataMessage->getSensitiveData()->serialize()),
+                    'sensitive_data' => json_encode((object) $sensitiveDataMessage->getSensitiveData()->serialize()),
                 ]);
             }
             $this->connection->commit();
         } catch (CoreException $e) {
             $this->connection->rollBack();
-            throw new RuntimeException('An exception occurred while saving sensitive data', 0, $e);
+            throw new RuntimeException('An exception occurred while appending sensitive data', 0, $e);
+        }
+    }
+
+    /**
+     * @param SensitiveDataMessageStream $sensitiveDataMessageStream
+     * @return void
+     */
+    public function update(SensitiveDataMessageStream $sensitiveDataMessageStream)
```

Comment:
`modify` rather than `update`? Seems to be a bit more in line with `append`. Dunno, just a thought.